rmw_fastrtps_cpp/include/rmw_fastrtps_cpp/TypeSupport_impl.h

```diff
@@ -513,7 +574,18 @@ void deserialize_array<std::string>(
 {
     (void)call_new;
     if (member->array_size_ && !member->is_upper_bound_) {
-        deser.deserializeArray(((rosidl_generator_c__String*)field)->data, member->array_size_);
+        auto deser_field = (rosidl_generator_c__String*)field;
+        std::vector<std::string> stringseq(member->array_size_);
```

Comment:
Forget my comments about this line. I was thinking about `field` and not `stringseq`. Their code really deserializes into `std::string` and then we copy it again from there into a c string. Not efficient but ok for now.